js/main.js

```diff
@@ -1,45 +1,491 @@
+  },
+  cruiser: {
+    name: "cruiser",
+    length: 3,
+    health: 3,
+    key: "player-cr",
+  },
+  firstDestroyer: {
+    name: "First Destroyer",
+    length: 3,
+    health: 3,
+    key: "player-fd",
+  },
+  secondDestroyer: {
+    name: "Second Destroyer",
+    length: 3,
+    health: 3,
+    key: "player-sd",
+  },
+  firstSubmarine: {
+    name: "First Submarine",
+    length: 2,
+    health: 2,
+    key: "player-fs",
+  },
+  secondSubmarine: {
+    name: "Second Submarine",
+    length: 2,
+    health: 2,
+    key: "player-ss",
+  },
+};
+let cpuShips = {
+  aircraftCarrier: {
+    length: 5,
+    health: 5,
+    key: "cpu-ac",
+  },
+  battleship: {
+    length: 4,
+    health: 4,
+    key: "cpu-bs",
+  },
+  cruiser: {
+    length: 3,
+    health: 3,
+    key: "cpu-cr",
+  },
+  firstDestroyer: {
+    length: 3,
+    health: 3,
+    key: "cpu-fd",
+  },
+  secondDestroyer: {
+    length: 3,
+    health: 3,
+    key: "cpu-sd",
+  },
+  firstSubmarine: {
+    length: 2,
+    health: 2,
+    key: "cpu-fs",
+  },
+  secondSubmarine: {
+    length: 2,
+    health: 2,
+    key: "cpu-ss",
+  },
+};
+const BASE_PLAYER_SHIP_STATES = JSON.parse(JSON.stringify(playerShips));
+const BASE_CPU_SHIP_STATES = JSON.parse(JSON.stringify(cpuShips));
+const LENGTH = CPU_BOARD[0].length;
+const HEIGHT = CPU_BOARD.length;
+const PLAYER_DIVS = [];
+const CPU_DIVS = [];
+const LOCATIONS = new Set();
+const playAgainBtn = document.createElement("button");
+/*----- variables -----*/
+let turn = 1;
+let winner = null;
+let lastHitColumn = null;
+let lastHitRow = null;
+let suitableIndex = null;
+let verticalCheckbox = false;
+let shipCounter = 0;
 /*----- cached element references -----*/
-const playerBoardSection = document.querySelector(".player-board-section");
-const cpuBoardSection = document.querySelector(".cpu-board-section");
+const PLAYER_BOARD_SECTION = document.querySelector(".player-board-section");
+const CPU_BOARD_SECTION = document.querySelector(".cpu-board-section");
+const activePlayerEl = document.querySelector("h2");
+const verticalCheckboxEl = document.querySelector("input");
+const verticalLabelEl = document.querySelector("label");
 /*----- event listeners -----*/
+CPU_BOARD_SECTION.addEventListener("click", handleMove);
+PLAYER_BOARD_SECTION.addEventListener("click", playerMouseClick);
+playAgainBtn.addEventListener("click", resetGame);
+verticalCheckboxEl.addEventListener("change", (e) => {
+  if (e.target.checked) {
+    verticalCheckbox = true;
+  } else {
+    verticalCheckbox = false;
+  }
+});
 /*----- functions -----*/
 render();
 
+//Initial rendering of the divs and header.
 function render() {
   createBoard("player");
   createBoard("cpu");
+  renderShips(CPU_BOARD);
+  activeTurn();
+}
+
+//Resetting the game to its base state after the match has ended
+function resetGame() {
+  playerShips = JSON.parse(JSON.stringify(BASE_PLAYER_SHIP_STATES));
+  cpuShips = JSON.parse(JSON.stringify(BASE_CPU_SHIP_STATES));
+  turn = 1;
```

Comment:
What does `1` mean in this scenario? Bit ambiguous. You should create a constant called `PLAYER_TURN` or something so that it's clear what this number represents